tests/test.linter.js

```diff
@@ -183,6 +183,38 @@ describe('Linter', function() {
       });
   });
 
+  it('Eslint ignore patterns and .eslintignorerc should be ignored', () => {
+    // Verify https://github.com/mozilla/addons-linter/issues/1288 is fixed
+    var addonLinter = new Linter({_: [
+      'tests/fixtures/webextension_node_modules_bower']});
+
+    // Stub print to prevent output.
+    addonLinter.print = sinon.stub();
+
+    assert.equal(addonLinter.collector.warnings.length, 0);
+    assert.equal(addonLinter.collector.warnings.length, 0);
+
+    var getFileSpy = sinon.spy(addonLinter, 'scanFile');
+
+    return addonLinter.scan()
+      .then(() => {
+        // Ensure no warnings are raised. If we would not overwrite eslint
+        // ignore patterns eslint ignores `node_modules` and
+        // `bower_components` by default. We want them to be scanned though.
+        assert.equal(addonLinter.collector.warnings.length, 0);
```

Comment:
I'll see if I can get sinon mocking up to spy into eslint directly but the idea here is, without the new configuration eslint actually throws a warning that the folder is ignored and we assume in the test, if that warning isn't thrown that the folder is scanned. It's a weird way of testing that, I agree.

I think I'll go with the error-approach, raising a error in the node_modules and bower_components folders but keep the test that there are no warnings being raised (thus the folders by default being ignored)